dom-event-delegation/main.ts

```diff
@@ -0,0 +1,19 @@
+const taskList = document.querySelector('.task-list');
+
+if (!taskList) throw new Error('the .task-list query failed');
+
+taskList.addEventListener('click', (event: Event) => {
+  const eventTarget = event.target as HTMLElement;
+
+  console.log('eventTarget: ', eventTarget);
+  console.log('tagName ', eventTarget.tagName);
+
+  if (eventTarget.tagName === 'BUTTON') {
+    const taskListItem = eventTarget.closest('.task-list-item');
+
+    console.log('Closest .task-list-item ancestor: ', taskListItem);
+
+    taskListItem?.remove();
+    /* I had to chat GPT this lst line for the '?' because I could not figure out what to do about the possibly null error */
```

Comment:
The '?' operator does it, but you could also do an old-school conditional if stmt:
if (taskListItem != null) { .... }

or 

if (taskListItem) { ... }